src/integrations/unixIntegrationManager.ts

```diff
@@ -67,15 +67,35 @@ export default class UnixIntegrationManager implements IntegrationManager {
   }
 
   protected async ensureIntegrationSymlinks(desiredPresent: boolean): Promise<void> {
-    // get list of integrations in the resources directory
-    const integrationNames = await fs.promises.readdir(this.resourcesDir);
+    const validIntegrationNames = await fs.promises.readdir(this.resourcesDir);
+    let currentIntegrationNames: string[] = [];
+
+    // integration directory may or may not be present; handle error if not
+    try {
+      currentIntegrationNames = await fs.promises.readdir(this.integrationDir);
+    } catch (error: any) {
+      if (error.code !== 'ENOENT') {
+        throw error;
+      }
+    }
+
+    // remove current integrations that are not valid
+    await Promise.all(currentIntegrationNames.map(async(name) => {
+      if (!validIntegrationNames.includes(name)) {
+        await fs.promises.rm(path.join(this.integrationDir, name), { force: true });
```

Comment:
I don't know if we should delete files that we don't recognize, even if it is "our" directory. Users might have put files there my accident ... I just think it's better to err on the side of caution when it comes to users' files.

Reply:
What if we stop bundling a utility? Without this, we are left with a dangling symlink. This hasn't happened yet, and I don't see how it would cause a problem, but it's nice to keep things clean.

@jandubois what do you think?

I don't have strong feelings either way.

Reply:
True, good point on assuming we own that directory and have the right to clean it up.